tensorflow/core/graph/mkl_graph_util.h

```diff
@@ -216,7 +224,18 @@ static inline bool IsMklNameChangeOp(const string& op_name, DataType T) {
     isTypeAllowed = (T == DT_COMPLEX128 || T == DT_COMPLEX64 ||
                      T == DT_DOUBLE || T == DT_FLOAT);
 #ifdef ENABLE_INTEL_MKL_BFLOAT16
-    isTypeAllowed = (isTypeAllowed || CheckBfloat16Support(T));
+    if (!isTypeAllowed) {
+      if (T == DT_BFLOAT16) {
+        if (IsBF16SupportedByOneDNNOnThisCPU()) {
+          isTypeAllowed = true;
+        } else {
+          // Restrict bfloat16 ops to platforms with at least AVX512 support,
+          // fall back to Eigen implementation otherwise.
+          BF16UnsupportedWarning();
+          isTypeAllowed = false;
```

Comment:
`isTypeAllowed` is already `false`. So this line is not necessary. Would you like to remove it? (I'm also fine leaving it in to save another round of testing, given that we are close to the branch cut deadline.)

Reply:
Yes, was double qualifying just to be sure this time around. :). Let us move on this time in the interest of getting the CI working internally and for the deadline. Will make a note internally for refactoring in future.

Reply:
Thank you for looking into this very soon! @penpornk